includes/class-wc-gateway-ppec-client.php

```diff
@@ -817,7 +817,7 @@ public function get_do_express_checkout_params( array $args ) {
 			'PAYMENTREQUEST_0_PAYMENTACTION' => $settings->get_paymentaction(),
 			'PAYMENTREQUEST_0_INVNUM'        => $settings->invoice_prefix . $order->get_order_number(),
 			'PAYMENTREQUEST_0_CUSTOM'        => json_encode( array(
-				'order_id'  => $order_id,
+				'order_id'  => apply_filters( 'woocommerce_order_number', $order_id ),
```

Comment:
If we want to pass the order number instead of the order id, why not just do `$order->get_order_number()` here?

I think this will cause issues, since we're reusing this custom variable in `WC_Gateway_PPEC_PayPal_Request_Handler::get_paypal_order`, where we get an order based on `order_id` (I assume that's the one we send here).

For example (assuming I have order ID 507):
```php
add_filter( 'woocommerce_order_number', function( $order_id ) {
	return 'Fancy order: ' . $order_id;
} );

add_action( 'init', function() {
	assert( false !== wc_get_order( 507 ) );
	assert( false !== wc_get_order( 'Fancy order: ' . 507 ) );
} );
```
Second assertion fails.

Reply:
Maybe we can send another custom parameter to indicate the actual order id, and re-use that.